src/vehicle.cpp

```diff
@@ -1018,10 +1018,15 @@ bool vehicle::can_unmount(int const p) const
         return false;
     }
 
+    //Can't remove an animal part if the animal is still contained
+    if( parts[p].has_flag( vehicle_part::animal_flag ) ) {
+        return false;
+    }
```

Comment:
Maybe add a message indicating why one can't remove the part in question?

Reply:
<whine>But that's hard.</whine>
 
Fine, I'll make the game better.

I accidentally amended the second commit in the series, so I had to undo that, and that broke the current repo history. If you pull into an existing branch, you'll get a merge error unless you reset first.